robotframework-ls/src/robotframework_ls/impl/flow_explorer_model_builder.py

```diff
@@ -37,29 +54,63 @@ def build_flow_explorer_model(completion_contexts: List[ICompletionContext]) ->
             }
             suites.append(suite)
 
-            for test_node_info in ast_utils.iter_tests(ast):
-                test_name = f"{suite_name}.{test_node_info.node.name}"
+            for test in ast_utils.iter_tests(ast):
+                test_name = f"{test.node.name} ({suite_name.lower()})"
                 test_body: list = []
-                test = {
+                test_info = {
                     "type": "task",
                     "name": test_name,
                     "doc": "",
                     "setup": None,
                     "teardown": None,
                     "body": test_body,
                 }
-                tasks.append(test)
-                for node_info in ast_utils.iter_all_nodes(
-                    test_node_info.node, recursive=False
-                ):
+                tasks.append(test_info)
+                for node_info in ast_utils.iter_all_nodes(test.node, recursive=False):
+                    recursion_stack.push(test_name)
                     _build_hierarchy(
                         completion_context,
                         node_info.stack,
                         node_info.node,
                         suite_name,
                         test_body,
                         {},
+                        recursion_stack,
+                        keywords,
                     )
+                    recursion_stack.pop()
+            for user_keyword in ast_utils.iter_keywords(ast):
+                user_keyword_name = f"{user_keyword.node.name} ({suite_name.lower()})"
+                user_keyword_body: list = []
+                user_keyword_info = {
+                    "type": "user-keyword",
+                    "name": user_keyword_name,
+                    "doc": "",
+                    "body": user_keyword_body,
+                }
+
+                # Keywords var will be populated when building hierarchy if importing statements
+                # Checking to see if it already exists before appending
+                for keyw in keywords:
```

Comment:
We can leave as is for now, but I think we could have a better approach here as looping to check for contains is not ideal (I can see it becoming something slow if we reach the 500+ keywords use case).

-- keywords could be some custom structure which has a that keywords list and an additional `set` with the name of the keywords added so far where we could just do a `user_keyword_name in container`.

p.s.: just noting, we should do this later not in this commit.